component/src/main/java/org/wso2/extension/siddhi/io/hl7/source/Hl7Source.java

```diff
@@ -0,0 +1,308 @@
+                Hl7Constants.DEFAULT_TLS_ENABLE));
+        this.tlsKeystoreFilepath = optionHolder.validateAndGetStaticValue(Hl7Constants.TLS_KEYSTORE_FILEPATH,
+                Hl7Constants.DEFAULT_TLS_KEYSTORE_FILEPATH);
+        this.tlsKeystorePassphrase = optionHolder.validateAndGetStaticValue(Hl7Constants.TLS_KEYSTORE_PASSPHRASE,
+                Hl7Constants.DEFAULT_TLS_KEYSTORE_PASSPHRASE);
+        this.conformanceProfileUsed = Boolean.parseBoolean(optionHolder.validateAndGetStaticValue(
+                Hl7Constants.HL7_CONFORMANCE_PROFILE_USED,
+                Hl7Constants.DEFAULT_CONFORMANCE_PROFILE_USED));
+        String profileFileName = optionHolder.validateAndGetStaticValue(Hl7Constants.HL7_CONFORMANCE_PROFILE_FILE,
+                Hl7Constants.DEFAULT_CONFORMANCE_PROFILE_FILE);
+        Hl7Utils.validateEncodingType(hl7Encoding, hl7AckEncoding, siddhiAppName, streamID);
+        this.hl7ReceivingApp = new Hl7ReceivingApp();
+        this.tlsKeystoreType = optionHolder.validateAndGetStaticValue(Hl7Constants.TLS_KEYSTORE_TYPE,
+                Hl7Constants.DEFAULT_TLS_KEYSTORE_TYPE);
+        if (conformanceProfileUsed) {
+            conformanceProfile = getConformanceProfile(profileFileName);
+        }
+        Hl7Utils.doTlsValidation(tlsEnabled, tlsKeystoreFilepath, tlsKeystorePassphrase, tlsKeystoreType,
+                siddhiAppName, streamID);
+    }
+
+    @Override
+    public Class[] getOutputEventClasses() {
+
+        return new Class[]{String.class};
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+
+        HapiContext hapiContext = new DefaultHapiContext();
+        MinLowerLayerProtocol mllp = new MinLowerLayerProtocol();
+        mllp.setCharset(charset);
+        hapiContext.setLowerLayerProtocol(mllp);
+        if (tlsEnabled) {
+            CustomCertificateTlsSocketFactory tlsFac = new CustomCertificateTlsSocketFactory(tlsKeystoreType,
+                    tlsKeystoreFilepath, tlsKeystorePassphrase);
+            hapiContext.setSocketFactory(new HapiSocketTlsFactoryWrapper(tlsFac));
+        }
+        hl7Service = hapiContext.newServer(port, tlsEnabled);
+        try {
+            hl7Service.startAndWait();
+        } catch (InterruptedException e) {
+            throw new ConnectionUnavailableException("Error occurred while starting the server on port: " + port
+                    + ", ", e);
+        }
+        hl7Service.registerApplication(new RegistrationEventRouting(), new Hl7ReceivingApp(sourceEventListener,
+                siddhiAppName, streamID, hl7Encoding, hl7AckEncoding, hapiContext, conformanceProfileUsed,
+                conformanceProfile));
+        hl7Service.setExceptionHandler(new Hl7ExceptionHandler());
+    }
+
+    @Override
+    public void disconnect() {
+
+        if (hl7Service != null) {
+            hl7Service.stop();
+        }
+
+    }
+
+    @Override
+    public void destroy() {
+
```

Comment:
Shall we add a comment mentioning why this is empty? please do the same in other places as well